config.go

```diff
@@ -70,32 +70,38 @@ func (c LoggingConfig) extractor() observability.ContextExtractor {
 }
 
 // MetricsConfig describes how telemetry should be configured.
+// Scope and Tally are exclusive; choose one.
 type MetricsConfig struct {
+	// Metrics is a *"go.uber.org/net/metrics".Scope for recording stats.
+	// YARPC does not push these metrics; pushing metrics from the root is an
+	// external concern.
+	Metrics *metrics.Scope
 	// Tally scope used for pushing to M3 or StatsD-based systems. By
 	// default, metrics are collected in memory but not pushed.
 	Tally tally.Scope
 }
 
-func (c MetricsConfig) registry(name string, logger *zap.Logger) (*pally.Registry, context.CancelFunc) {
-	r := pally.NewRegistry(
-		pally.Labeled(pally.Labels{
-			"component":  _packageName,
-			"dispatcher": pally.ScrubLabelValue(name),
-		}),
-		// Also expose all YARPC metrics via the default Prometheus registry.
-		pally.Federated(prometheus.DefaultRegisterer),
-	)
+func (c MetricsConfig) scope(name string, logger *zap.Logger) (*metrics.Scope, context.CancelFunc) {
+	if c.Tally != nil && c.Metrics != nil {
```

Comment:
a nil Metrics.Scope is actually valid.  We don't need to panic here, we can just create a noop Metrics scope.